### ISSUE TITLE
Mock clock leaves lock taken

### ISSUE THREAD

**Issue by tv42** [yrashk-archived/bpxe-go#68]
This pattern repeats:

https://github.com/bpxe/bpxe/blob/0828554a14ebbdc2c38b7564443c0036b0f05ab0/pkg/clock/mock.go#L53-L59

```
	m.Lock()
	ch := make(chan time.Time, 1)
	if duration.Nanoseconds() <= 0 {
		ch <- m.now
		close(ch)
		return ch
	}
```

This leaves the lock taken. Should use

```
m.Lock()
defer m.Unlock()
```

**Comment by yrashk** [comment]
You are right, thank you! `Until` does unlock in a similar branch, but `After` doesn't. Good spot.

In this case, I was avoiding using deferral based on some findings that deferrals are slow (and I was fighting slow tests using somebody else's mock clock) -- can't find the exact link right now. Actually, would be great to re-test this finding.

**Comment by tv42** [comment]
Reasonable use of deferrals was made fast in Go1.14, https://golang.org/doc/go1.14#runtime and every user of Go should be at least on go1.15 by now.

**Comment by yrashk** [comment]
> Reasonable use of deferrals was made fast in Go1.14, https://golang.org/doc/go1.14#runtime and every user of Go should be at least on go1.15 by now.

Thanks, good to know, I guess that part is outdated now, so I can improve code quality by avoiding unlock in branches. Thank you!

**Comment by yrashk** [comment]
Resolved in #70 and #72
